c7n/resources/elasticsearch.py

```diff
@@ -116,6 +116,60 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'EncryptionAtRestOptions.KmsKeyId'
 
 
+@ElasticSearchDomain.filter_registry.register('cross-cluster-search-connections')
+class ElasticSearchSearchConnections(ValueFilter):
+    """Check for Cloudfront distribution config values
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: elasticsearch-cross-cluster-search-connections
+            resource: aws.elasticsearch
+            filters:
+              - type: cross-cluster-search-connections
+                key: outbound[].SourceDomainInfo.OwnerId
```

Comment:
implementation looks good, wondering if we should adjust the spelling in policies though to take off the outer array element, what do you think of this

```yaml
type: cross-cluster
inbound:
   key: SourceDomainInfo.OwnerId
   value: {account_id}
   op: ne
outbound:
   key: DestinationDomainInfo.OwnerId
   value: {account_id}
   op: ne
```

Reply:
That definitely looks better, but I'm running into problems with validating the policy since I believe custodian checks that key, value, and op are present in the outermost part of the policy. Do you know the best way around this, or would it require overwriting the validation?

![image](https://user-images.githubusercontent.com/16889692/104146989-9c764e00-539a-11eb-86c5-914784388fcc.png)

Reply:
you want to use composition instead of subclassing valuefilter, and construct the value filter instance inline, see security group ingress/egress as another other example.

Reply:
I see thanks, I gave that a try and believe it's working now